src/uucore/src/lib/mods/clap_localization.rs

```diff
@@ -120,167 +105,209 @@ pub fn handle_clap_error_with_exit_code(err: Error, util_name: &str, exit_code:
+    // For UnknownArgument, we need to preserve clap's built-in tips (like using -- for values)
+    // while still allowing localization of the main error message
+    let rendered_str = err.render().to_string();
+    let _lines: Vec<&str> = rendered_str.lines().collect();
+
+    if let Some(invalid_arg) = err.get(ContextKind::InvalidArg) {
+        let arg_str = invalid_arg.to_string();
+        // Get localized error word with fallback
+        let error_word = translate!("common-error");
+
+        let colored_arg = maybe_colorize(&arg_str, Color::Yellow);
+        let colored_error_word = maybe_colorize(&error_word, Color::Red);
+
+        // Print main error message with fallback
+        let error_msg = translate!(
+            "clap-error-unexpected-argument",
+            "arg" => colored_arg.clone(),
+            "error_word" => colored_error_word.clone()
+        );
+        eprintln!("{error_msg}");
+        eprintln!();
+
+        // Show suggestion if available
+        let suggestion = err.get(ContextKind::SuggestedArg);
+        if let Some(suggested_arg) = suggestion {
+            let tip_word = translate!("common-tip");
+            let colored_tip_word = maybe_colorize(&tip_word, Color::Green);
+            let colored_suggestion = maybe_colorize(&suggested_arg.to_string(), Color::Green);
+            let suggestion_msg = translate!(
+                "clap-error-similar-argument",
+                "tip_word" => colored_tip_word.clone(),
+                "suggestion" => colored_suggestion.clone()
+            );
+            eprintln!("{suggestion_msg}");
+            eprintln!();
+        } else {
+            // Look for other clap tips (like "-- --file-with-dash") that aren't suggestions
+            // These usually start with "  tip:" and contain useful information
+            for line in _lines.iter() {
+                if line.trim().starts_with("tip:") && !line.contains("similar argument") {
+                    eprintln!("{}", line);
+                    eprintln!();
+                }
+            }
+        }
 
-            if let Some(invalid_arg) = err.get(ContextKind::InvalidArg) {
-                let arg_str = invalid_arg.to_string();
+        // Show usage information for unknown arguments
+        let usage_key = format!("{util_name}-usage");
+        let usage_text = translate!(&usage_key);
+        let formatted_usage = crate::format_usage(&usage_text);
+        let usage_label = translate!("common-usage");
+        eprintln!("{}: {}", usage_label, formatted_usage);
+        eprintln!();
+        eprintln!("{}", translate!("common-help-suggestion"));
+
+        std::process::exit(exit_code);
+    } else {
+        // Generic fallback case
+        let error_word = translate!("common-error");
+        let colored_error_word = maybe_colorize(&error_word, Color::Red);
+        eprintln!("{colored_error_word}: unexpected argument");
+        std::process::exit(exit_code);
+    }
```

Comment:
Both blocks end with `std::process::exit(exit_code);` and thus I would move it outside.
```suggestion
    } else {
        // Generic fallback case
        let error_word = translate!("common-error");
        let colored_error_word = maybe_colorize(&error_word, Color::Red);
        eprintln!("{colored_error_word}: unexpected argument");
    }
    std::process::exit(exit_code);
```